Makefile

```diff
@@ -0,0 +1,144 @@
+# From https://raw.githubusercontent.com/vincentbernat/hellogopher/1cf9ebeeaea41d2e159e196f7d176d6b12ff0eb2/Makefile
+# Licensed under Creative Commons Zero v1.0 Universal (CC0)
+
+MODULE   = $(shell env GO111MODULE=on $(GO) list -m)
+DATE    ?= $(shell date +%FT%T%z)
+VERSION ?= $(shell git describe --tags --always --dirty --match=v* 2> /dev/null || \
+			cat .version 2> /dev/null || echo v0)
+PKGS     = $(or $(PKG),$(shell env GO111MODULE=on $(GO) list ./...))
+BIN      = bin
+
+GO      = go
+TIMEOUT = 15
+V = 0
+Q = $(if $(filter 1,$V),,@)
+M = $(shell [ "$$(tput colors 2> /dev/null || echo 0)" -ge 8 ] && printf "\033[34;1m▶\033[0m" || printf "▶")
+
+export GO111MODULE=on
+
+.PHONY: all
+all: check-license mocks deep-copy fmt lint test-coverage | $(BIN)
+
+# Tools
+
+$(BIN):
+	@mkdir -p $@
+$(BIN)/%: | $(BIN) ; $(info $(M) installing $(PACKAGE)...)
+	$Q if [ ! -z "INSTALL_ROOT" ]; then cd $(INSTALL_ROOT); fi; env GOBIN=$(abspath $(BIN)) $(GO) install $(PACKAGE)
+
+GOIMPORTS = $(BIN)/goimports
+$(BIN)/goimports: PACKAGE=golang.org/x/tools/cmd/goimports@latest
+
+REVIVE = $(BIN)/revive
+$(BIN)/revive: PACKAGE=github.com/mgechev/revive@latest
+
+GOCOV = $(BIN)/gocov
+$(BIN)/gocov: PACKAGE=github.com/axw/gocov/gocov@latest
+
+GOCOVXML = $(BIN)/gocov-xml
+$(BIN)/gocov-xml: PACKAGE=github.com/AlekSi/gocov-xml@latest
+
+GOTESTSUM = $(BIN)/gotestsum
+$(BIN)/gotestsum: PACKAGE=gotest.tools/gotestsum@latest
+
+ADDLICENSE = $(BIN)/addlicense
+$(BIN)/addlicense: PACKAGE=github.com/google/addlicense@latest
+
+# FIXME this might break in the future since Mockery does not officially support installing with go install
+MOCKERY = $(BIN)/mockery
+$(BIN)/mockery: PACKAGE=github.com/vektra/mockery/v2@v2.12.3
+
+# Note: deepcopy-gen contains replace directives and cannot be installed in module-aware mode,
+# instead it is checked out and installed from the checked-out directory.
+DEEPCOPY = $(BIN)/deepcopy-gen
+$(BIN)/deepcopy-gen: PACKAGE=k8s.io/gengo/examples/deepcopy-gen
+$(BIN)/deepcopy-gen: INSTALL_ROOT=./$(GENGO_DIR)/examples/deepcopy-gen
+
+GENGO_DIR = $(BIN)/gengo
+$(GENGO_DIR): ; $(info $(M) cloning Gengo...)
+	@mkdir -p $@
+	$Q git clone -q https://github.com/kubernetes/gengo.git $@
+	$Q cd $@; git checkout -q 4627b89bbf1b # v0.0.0-20220307231824-4627b89bbf1b
+
+# Tests
+
+TEST_TARGETS := test-bench test-short test-verbose test-race
+.PHONY: $(TEST_TARGETS) check test tests
+test-bench:   ARGS=-run=__absolutelynothing__ -bench=. ## Run benchmarks
+test-short:   ARGS=-short        ## Run only short tests
+test-verbose: ARGS=-v            ## Run tests in verbose mode with coverage reporting
+test-race:    ARGS=-race         ## Run tests with race detector
```

Comment:
Interesting to note: test-race reports about a dozen race conditions in the `client` package.

Reply:
Yeah it's not surprising as I already ran into a couple of them in #36 and #38 . We're only using the package for integration tests so we're not too worried but we might need to take a look at these issues at some point.